File: scripts/test_solutions/746_min_cost_climbing_stairs.js
var minCostClimbingStairs = function(cost) {
    let prev2 = 0, prev1 = 0;
    for (let i = 2; i <= cost.length; i++) {
        const curr = Math.min(prev1 + cost[i - 1], prev2 + cost[i - 2]);
        prev2 = prev1; prev1 = curr;
    }
    return prev1;
};

File: scripts/test_solutions/746_min_cost_climbing_stairs.c
int minCostClimbingStairs(int* cost, int costSize) {
    int prev2 = 0;
    int prev1 = 0;
    for (int i = 2; i <= costSize; i++) {
        int curr = (prev1 + cost[i - 1] < prev2 + cost[i - 2])
                   ? prev1 + cost[i - 1]
                   : prev2 + cost[i - 2];
        prev2 = prev1;
        prev1 = curr;
    }
    return prev1;
}
